src/bin/pg_autoctl/config.c

```diff
@@ -34,10 +35,13 @@ build_xdg_path(char *dst,
 			   const char *name)
 {
 	char filename[MAXPGPATH];
-	char *home = getenv("HOME");
-	char *xdg_topdir;
+	char home[MAXPGPATH];
+	int homePathLength = 0;
+	char xdg_topdir[MAXPGPATH];
+	int xdg_topdir_length = 0;
 
-	if (home == NULL)
+	homePathLength = get_env_variable("HOME", home, MAXPGPATH);
+	if (homePathLength <= 0 || homePathLength >= MAXPGPATH)
```

Comment:
This is a behavioral change, since `HOME=""` will now fail (maybe this is fine)

Reply:
home is used later. it better be not empty